src/telemetry/Manager.cc

```diff
@@ -38,6 +53,449 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
+		int64_t observations = 0;
+
+		for ( const auto& b : histogram_data.buckets )
+			{
+			observations += b.count;
+			values_vec->Append(zeek::make_intrusive<zeek::DoubleVal>(b.count));
+			count_values_vec->Append(val_mgr->Count(b.count));
+			}
+
+		r->Assign(values_idx, values_vec);
+		r->Assign(count_values_idx, count_values_vec);
+
+		r->Assign(sum_idx, zeek::make_intrusive<zeek::DoubleVal>(histogram_data.sum));
+		r->Assign(observations_idx, zeek::make_intrusive<zeek::DoubleVal>(observations));
+
+		r->Assign(count_sum_idx, val_mgr->Count(histogram_data.sum));
+		r->Assign(count_observations_idx, val_mgr->Count(observations));
+		}
+
+	// This is a bit annoying. The bounds / upper bounds of the individual
+	// buckets are not accessible through the family hdl. We set the
+	// opts$bounds field based on bounds collected from an individual
+	// metric instances.
+	//
+	// As the opts_record is cached by the manager, this should only
+	// happen once.
+	//
+	// This is probalby working around something that should be fixed
+	// in the broker::telemetry level.
+	static auto opts_rt = zeek::id::find_type<zeek::RecordType>("Telemetry::MetricOpts");
+	static auto opts_rt_off_bounds = opts_rt->FieldOffset("bounds");
+	static auto opts_rt_off_count_bounds = opts_rt->FieldOffset("count_bounds");
+	static auto bounds_default_val = opts_rt->FieldDefault(opts_rt_off_bounds);
+
+	auto bounds_field_val = opts_record->GetField(opts_rt_off_bounds);
+	if ( ! bounds_field_val || bounds_field_val == bounds_default_val )
+		{
+		std::visit(
+			[&opts_record](const auto& histogram_data)
+			{
+				auto bounds_vec = make_intrusive<zeek::VectorVal>(double_vec_type);
+				for ( const auto& b : histogram_data.buckets )
+					bounds_vec->Append(zeek::make_intrusive<zeek::DoubleVal>(b.upper_bound));
+
+				opts_record->Assign(opts_rt_off_bounds, bounds_vec);
+
+				// If this is an int64_t histogram, also fill count_bounds
+				using T = std::decay_t<decltype(histogram_data)>;
+				if constexpr ( std::is_same_v<T, IntHistogramData> )
+					{
+					auto count_bounds_vec = make_intrusive<zeek::VectorVal>(count_vec_type);
+					for ( const auto& b : histogram_data.buckets )
+						count_bounds_vec->Append(val_mgr->Count(b.upper_bound));
+
+					opts_record->Assign(opts_rt_off_count_bounds, count_bounds_vec);
+					}
+			},
+			histogram);
+		}
+
+	return r;
+	}
+
+/**
+ * Encapsulate matching of prefix and name against a broker::telemetry::metric_family_hdl
+ */
+class MetricFamilyMatcher
+	{
+public:
+	MetricFamilyMatcher(std::string_view prefix, std::string_view name)
+		: prefix_pattern(prefix), name_pattern(name)
+		{
+		}
+
+	/**
+	 * @return true if the given family's prefix and name match, else false;
+	 */
+	bool operator()(const broker::telemetry::metric_family_hdl* family)
+		{
+		std::string prefix(broker::telemetry::prefix(family));
+		std::string name(broker::telemetry::name(family));
```

Comment:
Using regular parens sometimes don't do what you might expect ("most vexing parse"). It's generally a good rule to avoid them:

```suggestion
		auto prefix = std::string{broker::telemetry::prefix(family)};
		auto name = std::string{broker::telemetry::name(family)};
```

(You also just change the paren types, but using `auto` consistently also makes sure variables are always initialized.)

Reply:
Thanks!